lib/processors/themeBuilderWorker.js

```diff
@@ -0,0 +1,281 @@
+ */
+export class FsMainThreadInterface {
+	#comPorts = new Set();
+	#fsInterfaceReader = null;
+	#cache = Object.create(null);
+
+	/**
+	 * Constructor
+	 *
+	 * @param {@ui5/fs/fsInterface} fsInterfaceReader Reader for the Resources
+	 */
+	constructor(fsInterfaceReader) {
+		if (!fsInterfaceReader) {
+			throw new Error("fsInterfaceReader is mandatory argument");
+		}
+
+		this.#fsInterfaceReader = fsInterfaceReader;
+	}
+
+	/**
+	 * Adds MessagePort and starts listening for requests on it.
+	 *
+	 * @param {MessagePort} comPort port1 from a {code}MessageChannel{/code}
+	 */
+	startCommunication(comPort) {
+		if (!comPort) {
+			throw new Error("Communication channel is mandatory argument");
+		}
+
+		this.#comPorts.add(comPort);
+		comPort.on("message", (e) => this.#onMessage(e, comPort));
+		comPort.on("close", () => comPort.close());
+	}
+
+	/**
+	 * Ends MessagePort communication.
+	 *
+	 * @param {MessagePort} comPort port1 to remove from handling.
+	 */
+	endCommunication(comPort) {
+		comPort.close();
+		this.#comPorts.delete(comPort);
+	}
+
+	/**
+	 * Destroys the FsMainThreadInterface
+	 */
+	cleanup() {
+		this.#comPorts.forEach((comPort) => comPort.close());
+		this.#cache = null;
+		this.#fsInterfaceReader = null;
+	}
+
+	/**
+	 * Handles messages from the MessagePort
+	 *
+	 * @param {object} e data to construct the request
+	 * @param {string} e.action Action to perform. Corresponds to the names of
+	 * 	the public methods of "@ui5/fs/fsInterface"
+	 * @param {string} e.fsPath Path of the Resource
+	 * @param {object} e.options Options for "readFile" action
+	 * @param {MessagePort} comPort The communication channel
+	 */
+	#onMessage(e, comPort) {
+		switch (e.action) {
+		case "readFile":
+			this.#doRequest(comPort, {action: "readFile", fsPath: e.fsPath, options: e.options});
+			break;
+		case "stat":
+			this.#doRequest(comPort, {action: "stat", fsPath: e.fsPath});
+			break;
+		}
+	}
+
+	/**
+	 * Requests a Resource from the "@ui5/fs/fsInterface" and sends it to the worker threads
+	 * via postMessage.
+	 *
+	 * @param {MessagePort} comPort The communication channel
+	 * @param {object} parameters
+	 * @param {string} parameters.action Action to perform. Corresponds to the names of
+	 * 	the public methods of "@ui5/fs/fsInterface" and triggers this method of the
+	 *	"@ui5/fs/fsInterface" instance.
+	 * @param {string} parameters.fsPath Path of the Resource
+	 * @param {object} parameters.options Options for "readFile" action
+	 */
+	async #doRequest(comPort, {action, fsPath, options}) {
+		const cacheKey = [fsPath, action].join("-");
+		if (!this.#cache[cacheKey]) {
+			this.#cache[cacheKey] = new Promise((res) => {
+				if (action === "readFile") {
+					this.#fsInterfaceReader.readFile(fsPath, options, (error, result) => res({error, result}));
+				} else if (action === "stat") {
+					this.#fsInterfaceReader.stat(fsPath, (error, result) => res({error, result}));
```

Comment:
(Future?) Optimization: I guess now we could do the `JSON.parse(JSON.stringify())` once per stat result right here and cache the object.

Reply:
Ah, nice catch! Thanks!